src/app/models/building_bucket_calculator.js

```diff
@@ -13,21 +16,19 @@ define([
     this._tobuckets = null;
   };
 
-  BuildingBucketCalculator.prototype.getScale = function() {
+  BuildingBucketCalculator.prototype.getScale = function () {
     const extent = this.toExtent();
     const maxBuckets = this.buckets - 1;
 
-    let scale = d3.scale.linear()
-                  .domain(extent)
-                  .rangeRound([0, maxBuckets]);
+    let scale = d3.scaleLinear().domain(extent).rangeRound([0, maxBuckets]);
```

Comment:
Only minor changes to this file to update d3 V3 syntax to d3 V4 since it's better supported/documented.